### ISSUE TITLE
Exception importing calendar

### ISSUE THREAD

**Issue by theanorak-keith** [mampfes/hacs_waste_collection_schedule#331]
I've cobbled together a new source for Richmondshire council and added it to my home assistant instance. The sensor entities I've configured show up more-or-less how I'd expect them to, but I get a notification that waste_collection_schedule could not be set up, and to check the logs. The logs say:

```
022-08-16 01:54:26.038 ERROR (MainThread) [homeassistant.loader] Unexpected exception importing platform custom_components.waste_collection_schedule.calendar
2022-08-16 01:54:26.105 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform waste_collection_schedule.calendar: Platform not found (Exception importing custom_components.waste_collection_schedule.calendar).
```

configuration.yaml relevant lines are:

```
# waste collection!
waste_collection_schedule:
  sources:
    - name: richmondshire_gov_uk
      args:
        uprn: [a number identifying my house]
      customize:
        - type: 140L GARDEN BIN
          alias: Green bin
        - type: 55L RECYCLING BOX
          alias: Recycling
        - type: 240L GREY RUBBISH BIN
          alias: Black bin
  day_switch_time: "18:00"


sensor:
  - platform: waste_collection_schedule
    name: Green bin
    count: 2
    types:
      - Green bin
  - platform: waste_collection_schedule
    name: Recycling
    count: 2
    types:
      - Recycling
  - platform: waste_collection_schedule
    name: Black bin
    count: 2
    types:
      - Black bin
  - platform: waste_collection_schedule
    name: All rubbish
    count: 6
```

Lastly, in case it's my cobbled-together-from-other-people's-code source that's at fault, here's richmondshire_gov_uk.py:

```
from datetime import datetime
import requests
from waste_collection_schedule import Collection  # type: ignore[attr-defined]
import json

TITLE = "Richmondshire (North Yorkshire)"
DESCRIPTION = "description goes here"
URL = "https://www.richmondshire.gov.uk/collectionCalendar"
TEST_CASES = {"test 1": {"uprn": 200001767082}, "test 2": {"uprn": 200001767078}, "test3": {"uprn": 200001767079 }}

ICONS = {
    "240L GREY RUBBISH BIN": "mdi:trash-can",
    "55L RECYCLING BOX": "mdi:recycle",
    "140L GARDEN BIN": "mdi:leaf",
}

class Source:
    def __init__(
        self, uprn
    ):  # argX correspond to the args dict in the source configuration
        self._uprn = uprn

    def fetch(self):
        r = requests.get(
            f"https://www.richmondshire.gov.uk/Umbraco/Api/MyAreaApi/GetBinRoundData?uprn={self._uprn}"
        )
#        print(r.text)
        ids = r.json()
#        print (len(ids))

        entries = []

        for id in ids:
            entries.append(
                Collection(
                    date=datetime.strptime(id["start"], "%Y-%m-%dT%H:%M:%S").date(),
                    t=id["title"],
                    icon=ICONS.get(id["title"]),
                )
            )
        return entries
```

**Comment by dt215git** [comment]
Seems to work okay for me.

```
- platform: waste_collection_schedule
  name: Green bin
  count: 2
  types:
    - Green bin
- platform: waste_collection_schedule
  name: Recycling
  count: 2
  types:
    - Recycling
- platform: waste_collection_schedule
  name: Black bin
  count: 2
  types:
    - Black bin
```

![image](https://user-images.githubusercontent.com/86194065/184918744-fd8604d7-88b6-4580-b112-71fcbf76b672.png)

**Comment by theanorak-keith** [comment]
Are you using the current version of HA? 2022.8.0?

**Comment by dt215git** [comment]
![image](https://user-images.githubusercontent.com/86194065/185220588-e5a162b4-2f60-4640-863e-eb6d778631ac.png)
and v1.25.0 of Waste Collection Schedule.

**Comment by dt215git** [comment]
Presumably your richmodshire_gov_uk script is working and you can see the collection details when looing at the sensors under Developer Tools | States?

Did you install WCS via HACS, or manually?

**Comment by theanorak-keith** [comment]
As far as I can tell the script is working. I have the sensors added to a card temporarily (I removed the 'All rubbish' sensor from the config posted originally and changed the count to 6 for each) and they all show up there. In DevTools/States, I can see:

![Screenshot 2022-08-18 at 12 21 33](https://user-images.githubusercontent.com/21162802/185383605-547046ae-0977-4dd0-af0a-3218de8f0ddc.png)

![Screenshot 2022-08-18 at 12 21 48](https://user-images.githubusercontent.com/21162802/185383633-ac060b68-597d-48b7-8450-abec90356183.png)

![Screenshot 2022-08-18 at 12 21 57](https://user-images.githubusercontent.com/21162802/185383644-d9e682c5-d7d9-4503-aee2-ca3ca20cb803.png)

I installed via HACS; I also added a Google calendar integration, just to see if there was a problem with the calendar component in my installation, but that seems to work just fine.

**Comment by mampfes** [comment]
Are you running HA on Python 3.9?
#346

**Comment by theanorak-keith** [comment]
> Are you running HA on Python 3.9? #346

Sorry for the delayed response -- am on holiday :)

I am indeed running 3.9(.2); I saw just a few moments ago that there was an update to WCS that mentioned python 3.9 so I backed up my source file and updated WCS, then re-added my source file and restarted HA — problem solved!

Thank you so much. Will figure out how to submit my source file for Richmondshire in the next few days.
